fix(pipeline): std::time::Instant not in scope under --features turmoil, breaking test compilation

## Bug

PR #1254 fixed turmoil time simulation by conditionally importing `std::time::Instant` only when the `turmoil` feature is **not** active:

```rust
// crates/logfwd/src/pipeline.rs:13-14
#[cfg(not(feature = "turmoil"))]
use std::time::Instant;
```

However, several unit tests in the same file (under `#[cfg(test)] mod tests`) use bare `Instant::now()` via `use super::*` **without** being gated on `not(feature = "turmoil")`. For example:

- `test_machine_clean_shutdown_no_in_flight` (line ~2739)
- `test_machine_initialized_on_construction` (line ~2796)
- Other tests calling `Instant::now()` at lines ~2863, ~2924, ~2987, ~3025

When the `turmoil` feature is enabled, `std::time::Instant` is not imported into the module scope. `use super::*` pulls only what is in scope at the time, so these tests fail to compile:

```
cargo test -p logfwd --features turmoil --lib
error[E0425]: cannot find value `Instant` in this scope
```

## Fix

Either:
1. Add `use std::time::Instant;` inside the `mod tests` block (it's wall-clock time for test deadlines, not simulated time — this is correct).
2. Gate each affected test with `#[cfg(not(feature = "turmoil"))]` if they can't run under turmoil.

The simplest fix is option 1.

## Introduced by

PR #1254 (`fix(turmoil): use tokio::time::Instant for simulation`)

--- a/crates/logfwd/src/pipeline.rs
+++ b/crates/logfwd/src/pipeline.rs
@@ -1424,6 +1424,9 @@ fn now_nanos() -> u64 {
 mod tests {
     use super::*;
     use std::sync::atomic::Ordering;
+    // std::time::Instant is cfg-gated in the parent module for turmoil compatibility,
+    // but tests need it for timeout deadlines regardless of the turmoil feature flag.
+    use std::time::Instant;
 
     use logfwd_config::{Format, OutputConfig, OutputType};
     use logfwd_io::diagnostics::ComponentStats;

